main.tf

```diff
@@ -1,30 +1,31 @@
 resource "aws_elasticache_replication_group" "redis" {
-  engine = "redis"
+  engine = var.global_replication_group_id == null ? "redis" : null
 
-  parameter_group_name = aws_elasticache_parameter_group.redis.name
+  parameter_group_name = var.global_replication_group_id == null ? aws_elasticache_parameter_group.redis.name : null
   subnet_group_name    = aws_elasticache_subnet_group.redis.name
   security_group_ids   = concat(var.security_group_ids, [aws_security_group.redis.id])
 
   availability_zones    = var.availability_zones
-  replication_group_id  = "${var.name_prefix}-redis"
+  replication_group_id  = var.global_replication_group_id == null ? "${var.name_prefix}-redis" : "${var.name_prefix}-redis-replica"
   number_cache_clusters = var.cluster_mode_enabled ? null : var.number_cache_clusters
-  node_type             = var.node_type
+  node_type             = var.global_replication_group_id == null ? var.node_type : null
 
-  engine_version = var.engine_version
+  engine_version = var.global_replication_group_id == null ? var.engine_version : null
   port           = var.port
 
   maintenance_window         = var.maintenance_window
   snapshot_window            = var.snapshot_window
   snapshot_retention_limit   = var.snapshot_retention_limit
   final_snapshot_identifier  = var.final_snapshot_identifier
-  automatic_failover_enabled = var.automatic_failover_enabled && var.number_cache_clusters > 1 ? true : false
+  automatic_failover_enabled = var.global_replication_group_id == null && var.automatic_failover_enabled && var.number_cache_clusters > 1 ? true : false
```

Comment:
I tested this without the change to the way `automatic_failover_enabled` was originally written. Is there a particular reason why you need to include `var.global_replication_group_id` as part of the condition?

Reply:
In that case, can you please remove that change?

Reply:
```suggestion
  automatic_failover_enabled = var.automatic_failover_enabled && var.number_cache_clusters > 1 ? true : false
```